### PR TITLE
Taha/initial shape in handle change

### PR DESCRIPTION
**Pull request by Taha-Hassan-Git**
This PR passes the initial shape to the onHandleChange method. It makes it a bit easier to work with handles in custom shapes.

### Change Type

- [ ] `patch` — Bug fix
- [ ] `minor` — New feature
- [ ] `major` — Breaking change
- [ ] `dependencies` — Changes to package dependencies[^1]
- [ ] `documentation` — Changes to the documentation only[^2]
- [ ] `tests` — Changes to any test code only[^2]
- [x] `internal` — Any other changes that don't affect the published package[^2]
- [] I don't know

[^1]: publishes a `patch` release, for devDependencies use `internal`
[^2]: will not publish a new version

### Test Plan

- [ ] Unit Tests
- [ ] End to end tests

### Release Notes

- Add a brief release note for your PR here.

### CODE REVIEW

File: packages/tldraw/src/lib/utils/buildFromV1Document.ts [hunk 1]
```diff
@@ -553,8 +553,8 @@ export function buildFromV1Document(editor: Editor, document: LegacyTldrawDocume
 								})
 
 								const handles = editor.getShapeHandles(v2ShapeFresh)!
-
-								const change = util.onHandleChange!(v2ShapeFresh, {
+								const initial = undefined
```
Comment (Taha-Hassan-Git) [inline packages/tldraw/src/lib/utils/buildFromV1Document.ts:556]: I wasn't too sure what this file was, this seemed the best way to deal with not having access to the initial shape. 🤷🏽‍♂️

Comment (ds300) [reply packages/tldraw/src/lib/utils/buildFromV1Document.ts:556]: should be fine. This file is for loading old tldraw v1 file data into the new v2 format.

**Review by ds300 — COMMENTED** [review]
Yeah these are useful changes! Nice one.

File: packages/editor/src/lib/editor/shapes/ShapeUtil.ts [hunk 2]
```diff
@@ -588,6 +589,7 @@ export type TLOnChildrenChangeHandler<T extends TLShape> = (shape: T) => TLShape
 
 /** @public */
 export type TLOnHandleChangeHandler<T extends TLShape> = (
+	initial: T | undefined,
```
[the remark this answers is not in the document]
Comment (Taha-Hassan-Git) [reply packages/editor/src/lib/editor/shapes/ShapeUtil.ts:592]: Should I make it an optional property too?

**Pull request merged**